### PR TITLE
fix: also add `X-OC-Mtime` header to final chunks move

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud-libraries/nextcloud-upload/issues/1032

### CODE REVIEW

**Review by susnux — APPROVED** [review]

File: lib/uploader.ts [hunk 1]
```diff
@@ -242,6 +242,8 @@ export class Uploader {
 						method: 'MOVE',
 						url: `${tempUrl}/.file`,
 						headers: {
+							'X-OC-Mtime': file.lastModified / 1000,
+							'OC-Total-Length': file.size,
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply lib/uploader.ts:246]: I saw that @artonge also added this. I figured that while `X-OC-Mtime` was at stake here, maybe the same problem was also applying to the total lengh?
And bingo! It was, we also need it for the `MOVE`

https://github.com/nextcloud/server/blob/1719440079926ddd8fcbbc5bad76fda3fb21efad/apps/dav/lib/Upload/ChunkingPlugin.php#L58-L75
https://github.com/nextcloud/server/blob/1719440079926ddd8fcbbc5bad76fda3fb21efad/apps/dav/lib/Upload/ChunkingPlugin.php#L118-L119

[the remark this answers is not in the document]
Comment (susnux) [reply lib/uploader.ts:246]: (while it is not mentioned in the docs: https://docs.nextcloud.com/server/latest/developer_manual/client_apis/WebDAV/chunking.html#chunked-upload-v2 )

[the remark this answers is not in the document]
Comment (susnux) [reply lib/uploader.ts:246]: > And bingo! It was, we also need it for the `MOVE`

@skjnldsv that is the legacy chunking upload, so if we use v2 we do not need it :wink:

[the remark this answers is not in the document]
Comment (skjnldsv) [reply lib/uploader.ts:246]: > > And bingo! It was, we also need it for the `MOVE`
> 
> @skjnldsv that is the legacy chunking upload, so if we use v2 we do not need it 😉

Nope, the chunkingV2 is used for Object storage only :)

**Pull request merged**